aviatrix/resource_aviatrix_remote_syslog.go

```diff
@@ -0,0 +1,180 @@
+				Description: "Server Public Certificate: Public certificate of the controller signed by the same CA",
+			},
+			"private_key_file_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Server Private Key: Private key of the controller that pairs with the public certificate",
+			},
+			"protocol": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "TCP",
+				ValidateFunc: validation.StringInSlice([]string{"TCP", "UDP"}, false),
+				Description:  "Protocol: TCP or UDP (TCP by default)",
+			},
+			"template": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Optional Custom Template: Useful when forwarding to 3rd party servers like Datadog or Sumo",
+			},
+			"excluded_gateways": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "List of excluded gateways.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"notls": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if not protected by TLS.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Enabled or not.",
+			},
+		},
+	}
+}
+
+func resourceAviatrixRemoteSyslogCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	remoteSyslog := &goaviatrix.RemoteSyslog{
+		Server:            d.Get("server").(string),
+		Port:              d.Get("port").(int),
+		Protocol:          d.Get("protocol").(string),
+		Index:             d.Get("index").(int),
+		Template:          d.Get("template").(string),
+		CaCertificate:     d.Get("ca_certificate_file_path").(string),
+		PublicCertificate: d.Get("public_certificate_file_path").(string),
+		PrivateKey:        d.Get("private_key_file_path").(string),
+	}
+
+	if !((remoteSyslog.CaCertificate != "" && remoteSyslog.PublicCertificate != "" && remoteSyslog.PrivateKey != "") ||
+		(remoteSyslog.CaCertificate == "" && remoteSyslog.PublicCertificate == "" && remoteSyslog.PrivateKey == "")) {
+		return fmt.Errorf("one or more certificates missing")
+	}
+
+	var excludeGateways []string
+	for _, v := range d.Get("excluded_gateways").(*schema.Set).List() {
+		excludeGateways = append(excludeGateways, v.(string))
+	}
+	if len(excludeGateways) != 0 {
+		remoteSyslog.ExcludeGatewayInput = strings.Join(excludeGateways, ",")
+	}
+
+	if err := client.EnableRemoteSyslog(remoteSyslog); err != nil {
+		return fmt.Errorf("could not enable remote syslog: %v", err)
+	}
+
+	d.SetId("remote_syslog_" + strconv.Itoa(remoteSyslog.Index))
+	return nil
+}
+
+func resourceAviatrixRemoteSyslogRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*goaviatrix.Client)
+
+	server := d.Get("server").(string)
+
+	if server == "" {
+		id := d.Id()
+		log.Printf("[DEBUG] Looks like an import. Import Id is %s", id)
+
+		match, _ := regexp.Match("\\bremote_syslog_[0-9]\\b", []byte(id))
```

Comment:
This error should be checked if you want to use `regexp.Match`, however, the better way is to define a global variable like this 
```
var remoteSyslogMatcher = regexp.MustCompile("\\bremote_syslog_[0-9]\\b")
```
In Go, typically functions named `MustXXX` do not return error, instead they will panic if anything went wrong. In this case the function will panic if the regex wont compile, and since the variable is global scoped if you use an invalid regex it will crash on startup (making it easy to spot that the regex is wrong). 

Then with this `remoteSyslogMatcher` we can use it to check the ID format without having to worry about an error happening while compiling the regex. (also more efficient since we only compile the regex once)
```
match := remoteSyslogMatcher.Match([]byte(id))
```